### PR TITLE
Robot frame work

### PR DESCRIPTION
**Pull request by Jani-siv**
Base for embedded device debug feature, Robot framework modification and script for memory layout checking from elf file

### CODE REVIEW

File: arm/debugger/devices/Dumpper.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        data8.push_back(i & 0x00FF);
+        data8.push_back(i & 0xFF00);
+    }
+    DumpMemoryDataToFile(data8);
+}
+
+void Dumpper::DumpMemoryDataToFile(const std::vector<uint8_t>& data) const
+{
+
```
Comment (mezi-0) [inline arm/debugger/devices/Dumpper.cpp:33]: (33) Extra line can be removed

File: arm/debugger/devices/Dumpper.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        else if (counter+1 % 2 == 0)
+        {
+            ss << " ";
+        }
+        counter++;
+    }
+    std::string dumpData;
+    dumpData = ss.str();
+    WriteData(dumpData);
+}
```
Comment (mezi-0) [inline arm/debugger/devices/Dumpper.cpp:52]: (52-53) Additional line can be added

File: arm/debugger/devices/unit_tests/DumpperTests.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+namespace unittest {
+
+class DumpperTest : public testing::Test
+{
+public:
+    void SetUp() override
+    {}
+    void TearDown() override
+    {}
+
```
Comment (mezi-0) [inline arm/debugger/devices/unit_tests/DumpperTests.cpp:14]: (14) Extra line can be removed

File: arm/debugger/devices/unit_tests/MemoryTests.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+namespace unittest {
+
+class MemoryTest : public testing::Test
+{
+public:
+    void SetUp() override
+    {}
+    void TearDown() override
+    {}
+
```
Comment (mezi-0) [inline arm/debugger/devices/unit_tests/MemoryTests.cpp:15]: (15) Extra line can be removed

File: arm/debugger/devices/unit_tests/MmuTests.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+namespace unittest {
+
+class MmuTest : public testing::Test
+{
+public:
+    void SetUp() override
+    {}
+    void TearDown() override
+    {}
+
```
Comment (mezi-0) [inline arm/debugger/devices/unit_tests/MmuTests.cpp:15]: (15) Extra line can be removed

File: arm/debugger/devices/unit_tests/RegistersTests.cpp [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+namespace unittest {
+
+class RegistersTest : public testing::Test
+{
+public:
+    void SetUp() override
+    {}
+    void TearDown() override
+    {}
+
```
Comment (mezi-0) [inline arm/debugger/devices/unit_tests/RegistersTests.cpp:15]: (15) Extra line can be removed

File: arm/debugger/devices/unit_tests/SectionTests.cpp [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+namespace unittest {
+
+class SectionsTest : public testing::Test
+{
+public:
+    void SetUp() override
+    {}
+    void TearDown() override
+    {}
+
```
Comment (mezi-0) [inline arm/debugger/devices/unit_tests/SectionTests.cpp:15]: (15) Extra line can be removed

File: arm/debugger/main.cpp [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (mezi-0) [inline arm/debugger/main.cpp:1]: (1) Extra line can be removed

File: arm/src/cstart.c [hunk 9]
```diff
@@ -0,0 +1,18 @@
+#include <stdint.h>
+
+extern uint8_t _sbss;
+extern uint8_t _ebss;
+
+volatile uint8_t *data = &_sbss;
+volatile uint8_t *data_end = &_ebss;
+
+
```
Comment (mezi-0) [inline arm/src/cstart.c:9]: (9) Extra line can be removed

File: robot/FrameWorkTest.robot [hunk 10]
```diff
@@ -1,5 +1,7 @@
 *** Settings ***
 Library    ./Libraries/Framework.py
+Library           Process
+
```
Comment (mezi-0) [inline robot/FrameWorkTest.robot:4]: (4) Extra line can be removed

File: robot/Libraries/Framework.py [hunk 11]
```diff
@@ -8,3 +9,26 @@ def ping_local_host():
             return True
     return False
 
+
+def run_os_command(filename):
+    getmem = subprocess.Popen("../arm/scripts/memory_layout.sh " + filename, shell=True, stdout=subprocess.PIPE).stdout
+    mem = getmem.read()
+    val = mem.decode()
+    return val
+
```
Comment (mezi-0) [inline robot/Libraries/Framework.py:18]: (18) Extra line can be removed

**Comment by Jani-siv** [comment]
> Few line removing. Also in Dumpper.cpp can be added one additional line

done

**Review by mezi-0 — APPROVED** [review]

**Pull request merged**